app/modules/repo_manager/repo_manager.py

```
@@ -273,47 +521,704 @@ def _calculate_volume_bytes(self, path: str) -> Optional[int]:
+        bare_repo_path: Path,
+        ref: str,
+        auth_token: Optional[str],
+        repo_url: Optional[str] = None,
+    ) -> None:
+        """Fetch updates to bare repository for specific ref."""
+        try:
+            logger.info(f"Fetching ref '{ref}' for bare repo at {bare_repo_path}")
+
+            if auth_token and repo_url:
+                authenticated_url = self._build_authenticated_url(repo_url, auth_token)
+                subprocess.run(
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)